src/include/planner/index_scan_plan.h

```diff
@@ -20,11 +20,13 @@
 #include "common/types.h"
 #include "expression/abstract_expression.h"
 #include "storage/tuple.h"
+#include "index/scan_optimizer.h"
```

Comment:
Since there is an in-lined function that uses the class defined inside scan_optimizer it is hard to remove